model/src/main/java/com/javahelp/model/user/UserPassword.java

```diff
@@ -70,16 +69,13 @@ public void setSalt(byte[] salt) {
      * @return A concatenated array of salt and hashed password, with salt being up front.
      */
     private static byte[] combineByteArrays (byte[] salt, byte[] hash){
-        byte[] representation = new byte[salt.length + hash.length];
-        int position = 0;
-        for (byte element: salt){
-            representation[position] = element;
-            position++;
-        }
-        for (byte element: hash){
-            representation[position] = element;
-            position++;
+        if (salt.length == 0 || hash.length == 0) {
```

Comment:
Eventually, with a different constructor, this will no longer be possible; however, for now it makes sense.